CRM/Civicase/Form/Report/BaseExtendedReport.php

```diff
@@ -200,4 +221,404 @@ public function getAvailableJoins() {
+      if (isset($defaults['fields_defaults']) && in_array($spec['name'], $defaults['fields_defaults'])) {
+        $columns[$tableName]['metadata'][$fieldAlias]['is_fields_default'] = TRUE;
+      }
+
+      if (empty($spec['is_fields']) || (isset($options['fields_excluded']) && in_array($specName, $options['fields_excluded']))) {
+        $columns[$tableName]['fields'][$fieldAlias]['no_display'] = TRUE;
+      }
+
+      if (!empty($spec['is_filters']) && !empty($spec['statistics']) && !empty($options) && !empty($options['group_by'])) {
+        foreach ($spec['statistics'] as $statisticName => $statisticLabel) {
+          $columns[$tableName]['filters'][$fieldAlias . '_' . $statisticName] = array_merge($spec, [
+            'title' => E::ts('Aggregate filter : ') . $statisticLabel,
+            'having' => TRUE,
+            'dbAlias' => $tableName . '_' . $fieldAlias . '_' . $statisticName,
+            'selectAlias' => "{$statisticName}({$tableAlias}.{$spec['name']})",
+            'is_fields' => FALSE,
+            'is_aggregate_field_for' => $fieldAlias,
+          ]);
+          $columns[$tableName]['metadata'][$fieldAlias . '_' . $statisticName] = $columns[$tableName]['filters'][$fieldAlias . '_' . $statisticName];
+        }
+      }
+
+      foreach ($types as $type) {
+        if (!empty($spec['is_' . $type])) {
+          if ($type === 'join_filters') {
+            $fieldAlias = 'join__' . $fieldAlias;
+          }
+          $columns[$tableName][$type][$fieldAlias] = $spec;
+          if (isset($defaults[$type . '_defaults']) && isset($defaults[$type . '_defaults'][$spec['name']])) {
+            $columns[$tableName]['metadata'][$fieldAlias]['default'] = $defaults[$type . '_defaults'][$spec['name']];
+          }
+        }
+      }
+    }
+    $columns[$tableName]['prefix'] = isset($options['prefix']) ? $options['prefix'] : '';
+    $columns[$tableName]['prefix_label'] = isset($options['prefix_label']) ? $options['prefix_label'] : '';
+    if (isset($options['group_title'])) {
+      $groupTitle = $options['group_title'];
+    }
+    else {
+
+      // We can make one up but it won't be translated....
+      $groupTitle = ucfirst(str_replace('_', ' ', str_replace('civicrm_', '', $tableName)));
+    }
+    $columns[$tableName]['group_title'] = $groupTitle;
+    return $columns;
+  }
+
+  public function formatDisplay(&$rows, $pager = TRUE) {
+    // set pager based on if any limit was applied in the query.
+    if ($pager) {
+      $this->setPager();
+    }
+
+    // unset columns not to be displayed.
+    foreach ($this->_columnHeaders as $key => $value) {
+      if (!empty($value['no_display'])) {
+        unset($this->_columnHeaders[$key]);
+      }
+    }
+
+    // unset columns not to be displayed.
+    if (!empty($rows)) {
+      foreach ($this->_noDisplay as $noDisplayField) {
+        foreach ($rows as $rowNum => $row) {
+          unset($this->_columnHeaders[$noDisplayField]);
```

Comment:
tbh, Not sure what the loop is doing(must be some logic particular to the base class), I only overrode the base function because I needed to inject the `adjustRowTotal` function.